src/progressbar/progressbar.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+angular.module('mgcrea.ngStrap.progressbar', [])
+.provider('$progressbar', function (){
+    var defaults = {
+      barType: '',
+      animate: function (){ return true;}
+    };
+
+    this.$get = function (){
+      return {
+        defaults: defaults
+      };
+    };
+  })
+  .directive('bsProgressbar', function ($progressbar){
+    return {
+      restrict: 'E',
+      transclude: true,
+      replace: true,
+      templateUrl: 'progressbar/progressbar.tpl.html',
+      scope:{
+        value: '=',
+        type: '@',
+        animate: '&'
+      },
+      link: function (scope, element, attr){
+        scope.type = scope.type || $progressbar.defaults.barType;
+        scope.animate = angular.isDefined(scope.animate()) ? scope.animate : $progressbar.defaults.animate;
+        scope.$watch('type', function (){
+          if(scope.type) {
+            scope.barClass = 'progress-bar-' + scope.type;
```

Comment:
barClass does not really needs to live on the scope, it should be concatenated in the `ng-class` inside the template